lld/ELF/ScriptParser.cpp

```diff
@@ -1719,19 +1719,20 @@ ScriptParser::readSymbols() {
   while (!errorCount()) {
     if (consume("}"))
       break;
-    if (consumeLabel("local")) {
-      v = &locals;
-      continue;
-    }
-    if (consumeLabel("global")) {
-      v = &globals;
-      continue;
-    }
 
     if (consume("extern")) {
       SmallVector<SymbolVersion, 0> ext = readVersionExtern();
       v->insert(v->end(), ext.begin(), ext.end());
     } else {
+      if (consume("local:") || (consume("local") && consume(":"))) {
```

Comment:
I think this introduces a bug that a symbol named `local` or `global` will not be accepted.

The "local" and "global" name check should be performed after `next()`.

Reply:
It makes a lot of sense now thank you very much! I just updated it again.